livekit-plugins/livekit-plugins-aws/livekit/plugins/aws/stt.py

```
@@ -0,0 +1,231 @@
+                enable_channel_identification=self._opts.enable_channel_identification,
+                number_of_channels=self._opts.number_of_channels,
+                enable_partial_results_stabilization=self._opts.enable_partial_results_stabilization,
+                partial_results_stability=self._opts.partial_results_stability,
+                language_model_name=self._opts.language_model_name,
+            )
+
+            # this function basically convert the queue into a async generator
+            async def input_generator():
+                try:
+                    async for frame in self._input_ch:
+                        if isinstance(frame, rtc.AudioFrame):
+                            await stream.input_stream.send_audio_event(
+                                audio_chunk=frame.data.tobytes()
+                            )
+                    await stream.input_stream.end_stream()
+
+                except Exception as e:
+                    logger.exception(f"an error occurred while streaming inputs: {e}")
+
+            async def handle_transcript_events():
+                try:
+                    async for event in stream.output_stream:
+                        if isinstance(event, TranscriptEvent):
+                            self._process_transcript_event(event)
+                except Exception as e:
+                    logger.exception(
+                        f"An error occurred while handling transcript events: {e}"
+                    )
+
+            tasks = [
+                asyncio.create_task(input_generator()),
+                asyncio.create_task(handle_transcript_events()),
+            ]
+            # try to connect
+            try:
+                await asyncio.gather(*tasks)
+            finally:
+                await utils.aio.gracefully_cancel(*tasks)
+        except Exception as e:
+            logger.exception(f"An error occurred while streaming inputs: {e}")
```

I think this is swallowing exceptions? In this case the baseclass will not try to reconnect on failure